plugin/pulpcore/plugin/changeset/iterator.py

```diff
@@ -0,0 +1,238 @@
+import itertools
+
+from collections.abc import Iterable
+from logging import getLogger
+
+from django.db.models import Q
+
+from pulpcore.download import Download
+
+from .model import RemoteArtifact
+
+
+log = getLogger(__name__)
+
+
+class BatchIterator(Iterable):
+    """
+    Iterate large numbers of items in batches.
+
+    Attributes:
+        iterable (Iterable): An iterable to batch.
+        batch (int): The size of each batch.
+
+    Examples:
+        >>>
+        >>> numbers = [1, 2, 3, 4, 5, 6]
+        >>> for batch in BatchIterator(numbers, 3):
+        >>>     repr(batch)
+        '(1, 2, 3)'
+        '(4, 5, 6)'
+    """
+
+    # Default batch size.
+    BATCH = 1000
```

Comment:
This isn't really a constant, subclasses set batch size by overriding this value. How would you feel about `self.batch_size`?